plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/ProjectItem.java

```diff
@@ -55,6 +59,46 @@ public ProjectItem(TreeItem treeItem) {
 		project = new Project(new DefaultTreeItem(treeItem.getPath()[0]));
 		path = treeItem.getPath();
 	}
+	
+	/**
+	 * Runs project with the specified launcher.
+	 * 
+	 * @param launcher Launcher
+	 */
+	@SuppressWarnings("unchecked")
+	public void runAs(String launcher) {
+		select();
+		
+		Matcher<String> launcherMatcher = new WithMnemonicTextMatcher(new RegexMatcher("[0-9]* " + launcher));
+		new ContextMenu(new WithMnemonicTextMatcher("Run As"), new WithMnemonicTextMatcher(launcherMatcher)).select();
+	}
+	
+	/**
+	 * Runs project as JUnit Test and waits until it is finished.
+	 */
+	public void runAsJUnitTest() {
+		runAsJUnitTest(TimePeriod.LONG);
+	}
+	
+	/**
+	 * Runs project as JUnit Test waits until it is finished in a given timeout.
+	 * 
+	 * @param timeout Timeout
+	 */
+	public void runAsJUnitTest(TimePeriod timeout) {
```

Comment:
I don't think it is necessary to mention it because the logic is completely encapsulated into the method (the settings for console activation are set back at the end of the method).